### PR TITLE
Fix broadcast expand for the new return system

### PR DESCRIPTION
**Pull request by rmoyard**
**Description of the Change:**

The broadcast expand transform was stacking the results along the wrong axis, this is corrected.

**Example**
```
import pennylane as qml
import pennylane.numpy as np

x = 0.2
y = np.array([0.1, 5.1])

dev = qml.device("default.qubit", wires=2)


@qml.qnode(dev)
def circuit(x, y):
    qml.RX(x, wires=0)
    qml.RX(y, wires=1)
    return qml.expval(qml.PauliZ(0)), qml.expval(qml.PauliY(1))


print(circuit(x, y))
```

`(tensor([0.98006658, 0.98006658], requires_grad=True), tensor([-0.09983342,  0.92581468], requires_grad=True))`

### CODE REVIEW

File: pennylane/transforms/broadcast_expand.py [hunk 1]
```diff
@@ -112,7 +112,12 @@ def broadcast_expand(tape):
 
     def processing_fn(results):
         if len(tape.measurements) > 1 and qml.active_return():
-            results = [qml.math.hstack(r) for r in results]
+            processed_results = [None] * len(tape.measurements)
```
Comment (timmysilv) [inline pennylane/transforms/broadcast_expand.py:115]: why are you pre-allocating `None` values instead of just using a list-comprehension below?

File: pennylane/transforms/broadcast_expand.py [hunk 2]
```diff
@@ -112,7 +112,12 @@ def broadcast_expand(tape):
...
+            for i in range(len(tape.measurements)):
+                processed_results[i] = qml.math.squeeze(
+                    qml.math.stack([results[b][i] for b in range(tape.batch_size)])
```
Comment (timmysilv) [inline pennylane/transforms/broadcast_expand.py:118]: did `broadcast_expand` not account for `batch_size` in the old return system, or did it happen auto-magically because of the shape of returned data?

File: tests/devices/test_default_qubit_broadcasting.py [hunk 3]
```diff
@@ -1985,4 +1987,5 @@ def circuit(x, y):
 
         assert circuit.device.num_executions == len(y)
         tol = 1e-10 if shots is None else 1e-2
+        print(out, expected)
```
Comment (timmysilv) [inline tests/devices/test_default_qubit_broadcasting.py:1990]:
```suggestion
```

File: pennylane/transforms/broadcast_expand.py [hunk 4]
```diff
@@ -112,7 +112,12 @@ def broadcast_expand(tape):
...
+                )
+            return tuple(processed_results)
```
[the remark this answers is not in the document]
Comment (rmoyard) [reply pennylane/transforms/broadcast_expand.py:120]: Priorities:
1. The shot vector dimension (is a tuple-like dimension)
2. The enumeration of different return values (can be a tuple-like dimension) (measurements)
3. The broadcasting dimension
4. The "standard" return value shape (e.g. scalar for `expval`, 1D for `probs`, 2D for `sample`) (all tensor-like, except sample with `counts=True`)

File: pennylane/transforms/broadcast_expand.py (hunk 1 above)
Comment (rmoyard) [reply pennylane/transforms/broadcast_expand.py:115]: Good point, updated!

File: pennylane/transforms/broadcast_expand.py (hunk 2 above)
Comment (rmoyard) [reply pennylane/transforms/broadcast_expand.py:118]: Second one, there was no need for processing.

**Review by mudit2812 — APPROVED** [review]
Looks good. Thanks!
